flatdata-rs/lib/src/generator.rs

```diff
@@ -60,14 +49,39 @@ use std::{
 /// // re-export if desired
 /// pub use my_schema::*;
 /// ```
-pub fn generate(
-    output_dir: &Path,
-    schemas_path: &Path,
-    generator: &Path,
-) -> Result<(), GeneratorError> {
-    let output_dir = output_dir.canonicalize()?;
-    let schemas_path = schemas_path.canonicalize()?;
-    let generator = generator.canonicalize()?;
+pub fn generate(schemas_path: &str) -> Result<(), GeneratorError> {
+    let schemas_path = PathBuf::from(schemas_path).canonicalize()?;
+    let out_dir = PathBuf::from(env::var("OUT_DIR").expect("not running in a build.rs script?"));
+
+    // create a virtualenv in the target folder
+    eprintln!("creating virtualenv");
+    let _ = Command::new("python3")
+        .arg("-m")
+        .arg("venv")
+        .arg(&out_dir)
+        .stdout(Stdio::null())
+        .spawn()
+        .map_err(|e| GeneratorError::PythonError(e))?
+        .wait()?;
+
+    // install dependencies
+    eprintln!("installing PyPI dependencies");
+    let generator_path = if let Ok(path) = env::var("FLATDATA_GENERATOR_PATH") {
```

Comment:
Wouldn't it be better to have this envvar as function arguments? If yes, I guess, OUT_DIR can be also a param.

Reply:
fixed.